nat/service_ics.go

```diff
@@ -206,6 +219,57 @@ func (nat *serviceICS) getInternalInterfaceName() (string, error) {
 	return ifaceName, nil
 }
 
+func (nat *serviceICS) interfaceIPWatchdog() {
+	for {
+		select {
+		case <-time.After(5 * time.Second):
+			for ifaceName, rule := range nat.ifaces {
+				_, ipnet, err := net.ParseCIDR(rule.SourceAddress)
+				if err != nil {
+					log.Warnf("%s Failed to parse IP-address: %s", natLogPrefix, rule.SourceAddress)
+					continue
+				}
+
+				incrementIP(ipnet.IP)
+				if err := nat.setInterfaceAddr(ifaceName, ipnet); err != nil {
+					log.Warnf("%s Failed to assign IP-address '%s' to interface '%d'", natLogPrefix, ipnet.String(), ifaceName)
+				}
+			}
+		case <-nat.ipAddrWatchdog:
+			return
+		}
+	}
+}
+
+func (nat *serviceICS) setInterfaceAddr(name string, subnet *net.IPNet) error {
+	iface, err := net.InterfaceByName(name)
+	if err != nil {
+		return err
+	}
+
+	addrs, err := iface.Addrs()
+	if err != nil {
+		return err
+	}
+	for _, addr := range addrs {
+		if addr.String() == subnet.String() {
+			return nil // Interface address already assigned, nothing to do.
+		}
+	}
+
+	out, err := nat.powerShell("netsh interface ip set address name=\"" + name + "\" source=static " + subnet.String())
+	return errors.Wrap(err, string(out))
+}
+
+func incrementIP(ip net.IP) {
```

Comment:
- should return newly calculated value `incrementIP(ip IP) IP`
- or should accept IP reference `incrementIP(ipPtr *IP)` if you're changing value to new value

Reply:
It is pointer anyway. Changed it returns the new value to make it more clear.